### DISCUSSION TITLE
Backup Docker Volumes

### DISCUSSION THREAD

**Discussion by Rustypredator** [coollabsio/coolify#4597] [Feature Requests]
Simple: Let me use the S3 Storage to back up specific Volumes.  
I saw this Discussion: https://github.com/coollabsio/coolify/discussions/3660
And while the solution using duplicati or some other service would work, i would prefer an integrated solution within coolify.

**Comment by straatrakker** [comment]
Would love to see this as well.

**Comment by real-marty** [comment]
Would be veeery nice to have the option to backup it and restore 👀

**Comment by Copystrike** [comment]
This would be a super valuable feature! Would also like having a restore option as well would complete the workflow.

**Comment by marco-derek** [comment]
Highly Reccomend using this
https://hub.docker.com/r/offen/docker-volume-backup

Can be added to any existing compose service. Alternatively you can just add the docker volume directly if you know the direct volume path

**Comment by laravieira** [comment]
Also would love to see that. Right now is a Dokploy major advantage on Coolify.

**Comment by DavidRadobenko** [comment]
It would also be great to choose multiple S3 Storages to backup too, not just one like in Settings -> Backup -> Scheduled Backups or Scheduled Backups for databases in services.

**Comment by endrence3** [comment]
It would be great to see **native Docker Volume Backup support** in Coolify.  

For example, [Dokploy](https://dokploy.com/) supports this natively, which makes it a very production-ready solution. Services like **Vaultwarden** (with its default SQLite DB) benefit greatly from having native volume backups to S3 (ideally via **restic**).  

This feels like the **one major missing feature** in Coolify that makes it harder to choose as a complete PaaS solution. Adding it would truly make Coolify a one-stop platform. 🚀

**Comment by hendrikebbers** [comment]
Just coming from a recreation of a coolify instance with multiple app servers. Creating new DB instances and import Backups worked without any problem. Would love to see the same for storage :)

**Comment by philipthomasme** [comment]
IMHO this is the most important feature missing in Coolify.

**Comment by sancho1952007** [comment]
This is one major feature Coolify is missing compared to Dokploy. I really need this feature!

**Comment by TriHydera** [comment]
Also good in case of data loss

**Comment by hiyukoim** [comment]
+1! I want my apps to be portable. I want to back up everything and easily copy or move them to another server whenever I need. I can't wait to see this feature :)

**Reply by svk27** [reply]
I guess we can already clone and move resources, I guess as of Coolify Version 4.0.0-beta.474 you can simply go to your Resource > "Resource Operations" and you can Clone Resource or Move to other Project locally. But I get your point, I am after the same thing :)

**Comment by svk27** [comment]
+1, Only Major thing missing rn!

**Comment by Cinzya** [comment]
This is now being worked on: https://x.com/heyandras/status/2077388906056634452?s=20

**Comment by andrasbacsai** [comment]
It is implemented in the latest version.
